Modules/socketmodule.c

```diff
@@ -5432,39 +5443,16 @@ sock_initobj_impl(PySocketSockObject *self, int family, int type, int proto,
             proto = 0;
         }
 #ifdef MS_WINDOWS
-        /* Windows implementation */
-#ifndef WSA_FLAG_NO_HANDLE_INHERIT
-#define WSA_FLAG_NO_HANDLE_INHERIT 0x80
-#endif
-
         Py_BEGIN_ALLOW_THREADS
-        if (support_wsa_no_inherit) {
-            fd = WSASocketW(family, type, proto,
-                           NULL, 0,
-                           WSA_FLAG_OVERLAPPED | WSA_FLAG_NO_HANDLE_INHERIT);
-            if (fd == INVALID_SOCKET) {
-                /* Windows 7 or Windows 2008 R2 without SP1 or the hotfix */
-                support_wsa_no_inherit = 0;
-                fd = socket(family, type, proto);
-            }
-        }
-        else {
-            fd = socket(family, type, proto);
-        }
+        fd = WSASocketW(family, type, proto,
+                        NULL, 0,
+                        WSA_FLAG_OVERLAPPED | WSA_FLAG_NO_HANDLE_INHERIT);
```

Comment:
`WSA_FLAG_NO_HANDLE_INHERIT` is missing from an earlier call:

https://github.com/python/cpython/blob/63267d801bdeb28dfaae3036376be736bf43514a/Modules/socketmodule.c#L5348-L5351